internal/jobcontainers/jobcontainer.go

```diff
@@ -176,11 +177,38 @@ func (c *JobContainer) CreateProcess(ctx context.Context, config interface{}) (_
 
 	// Replace any occurences of the sandbox mount point env variable in the commandline.
 	// %CONTAINER_SANDBOX_MOUNTPOINT%\mybinary.exe -> C:\C\123456789\mybinary.exe
-	commandLine := c.replaceWithMountPoint(conf.CommandLine)
+	commandLine, _ := c.replaceWithMountPoint(conf.CommandLine)
+
+	removeDriveLetter := func(name string) string {
+		// If just the letter and colon (C:) then replace with a single backslash. Else just trim the drive letter and leave the rest of the
+		// path.
+		if len(name) == 2 && name[1] == ':' {
+			name = "\\"
+		} else if len(name) > 2 && name[1] == ':' {
+			name = name[2:]
+		}
+		return name
+	}
 
 	workDir := c.sandboxMount
 	if conf.WorkingDirectory != "" {
-		workDir = c.replaceWithMountPoint(conf.WorkingDirectory)
+		var changed bool
+		// For now, we join the working directory requested with where the sandbox volume is located. It's expected that the default behavior
+		// would be to treat all paths as relative to the volume.
+		//
+		// For example:
+		// A working directory of C:\ would become C:\C\12345678\
+		// A working directory of C:\work\dir would become C:\C\12345678\work\dir
+		//
+		// The below calls replaceWithMountPoint to replace any occurrences of the environment variable that points to where the container image
+		// volume is mounted.
+		workDir, changed = c.replaceWithMountPoint(conf.WorkingDirectory)
+		// If the working directory was changed, that means the user supplied %CONTAINER_SANDBOX_MOUNT_POINT%\\my\dir or something similar.
+		// In that case there's nothing left to do, as we don't want to join it with the mount point again.. If it *wasn't* changed, then we
+		// need to join it with the mount point, as it's some normal path.
+		if !changed {
+			workDir = filepath.Join(c.sandboxMount, removeDriveLetter(workDir))
+		}
```

Comment:
I don't really get this. You're taking the sandboxMount and joining it with the sandboxMount minus the drive letter? what's the point of that?

Reply:
`workDir` by this point would be whatever was in `conf.WorkingDirectory` as that's what `c.replaceWithMountpoint` returns. It takes the string given and returns a new string with any occurrences of $env:CONTAINER_SANDBOX_MOUNT_POINT or %CONTAINER_SANDBOX_MOUNTPOINT% replaced. If there were none, it returns the string as is and the changed bool will be false. The reason for checking if the string changed is that if someone explicitly had a env var replacement we don't want to append the same volume mount path on the working directory again. 

For example:
workdir: `replaceWithMountPoint($env:CONTAINER_SANDBOX_MOUNT_POINT\my\work\dir)` -> C:\C\123456789abcdef\my\work\dir and `changed` will be true so we will skip appending the volume path again. 

workdir: `replaceWithMountPoint(C:\my\work\dir)` -> C:\my\work\dir and changed will be false. Because of that we'll enter the `if !changed` condition and join `c.sandboxMount` with the working directory requested with the drive letter chopped off. This will ultimately end up the same as the first example.

Reply:
But for more context, we want the working directory (with the current approach for HostProcess/job containers) to always be inside the volume path. This will make things more familiar to normal container scenarios if an application is using relative paths. 

So if `my-config-file.json` was in a subdirectory foo in the root image which is located at C:\C\12345678 now, we want a working directory of C:\ and an application requesting foo\my-config-file.json to work. That currently won't work without this change as the user has no clue where the volume will be located before launching the container and we're not joining their requested path to the volume path, so we'd be looking for my-config-file.json at C:\foo\my-config-file.json.